docs/examples.md

```diff
@@ -1,9 +1,41 @@
 # Examples
 
-Here are some examples you can use as a model when writing your own type definitions.
+Here are some examples you can use as a model when writing your typings.
 
-## Single function
+There are a few kinds of source package:
 
+1. Package that is part of the environment.
+2. Package that should be loaded in a script tag.
+3. Package that written in CommonJs / NodeJs style
+  1. Package that also pollutes the global namespace
+4. Package that written in ES6+
+  1. Package that also pollutes the global namespace
+5. Package that written in TypeScript and compiled to JavaScript with declaration `.d.ts` files.
+
+For 1 and 2, you will create an ambient (global) typing.
+
+For 3, you will create an external module typing using `export =`.
+
+For 4, you will create an external module typing using ESM syntax (ES6/ES2015 default export and named export).
```

Comment:
ESM is the "standard" term for ES6 module as it has been pulled out as separate entity. (https://github.com/systemjs/systemjs/blob/c771e001ab900d9c47934f887284c2c6cac58e23/docs/module-formats.md)

But i'm fine with ES6 module syntax too, just try to be "accurate"